improver/spotdata/spot_extraction.py

```diff
@@ -114,25 +114,31 @@ def extract_coordinates(self, neighbour_cube: Cube) -> Cube:
         )
 
     @staticmethod
-    def extract_diagnostic_data(
-        coordinate_cube: Cube, diagnostic_cube: Cube
-    ) -> ndarray:
+    def extract_data(
```

Comment:
This method has got a bit complex and it took me a while to get my head around it. It might be better as two or three methods:

1. `get_xy_indices`
2. `get_diagnostic_data` (now only a couple of lines long, might not even be needed)
3. `get_coord_data`

Then you wouldn't have to have two different exit points with different return types.